src/librustc_mir/transform/const_prop.rs

```diff
@@ -407,18 +407,6 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
                 let arg = self.eval_operand(arg, source_info)?;
                 let val = self.use_ecx(source_info, |this| {
                     let prim = this.ecx.read_immediate(arg)?;
-                    match op {
-                        UnOp::Neg => {
```

Comment:
oh... we have this check for debug mode... I guess keep it but add an `opt-level` check here ( like https://github.com/rust-lang/rust/blob/35c9e5f1229d8b634ebd98c49e35c74e237e17f7/src/librustc_mir/transform/inline.rs#L42 ) so we only run this code in release mode (and add a comment so it's clear why we need to do this)

Reply:
Probably a good idea to do the same release mode opt out in https://github.com/rust-lang/rust/blob/35c9e5f1229d8b634ebd98c49e35c74e237e17f7/src/librustc_mir/transform/const_prop.rs#L476

Reply:
So, should we have the check like
```rust
self.tcx.sess.opts.debugging_opts.mir_opt_level >= 2
&& prim.to_bits()? == (1 << (prim.layout.size.bits() - 1)) {..}
```
here and L476?

Reply:
Yes

Reply:
pushed the change